test/integration/targets/rpm_key/tasks/rpm_key.yaml

```
@@ -138,6 +138,44 @@
   assert:
     that: "'rsa sha1 (md5) pgp md5 OK' in sl_check.stdout or 'digests signatures OK' in sl_check.stdout"
 
+- name: Issue 20325 - Verify fingerprint of key, invalid fingerprint - EXPECTED FAILURE
+  rpm_key:
+    key: "{{ role_path }}/files/RPM-GPG-KEY.dag.txt"
+    fingerprint: 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111
+  register: result
+  failed_when: result is success
+
+- name: Issue 20325 - Assert Verify fingerprint of key, invalid fingerprint
+  assert:
+    that:
+       - result is success
+       - result is not changed
+       - "'does not match the key fingerprint' in result.msg"
+
+- name: Issue 20325 - Verify fingerprint of key, valid fingerprint
+  rpm_key:
+    key: "{{ role_path }}/files/RPM-GPG-KEY.dag.txt"
```

This is only working because Ansible is being run in `local` mode in CI, so the control node and the managed node are the same. We plan to split this in the future, which means this test will fail.

This needs to be a path to the file on the managed node or a URL. I put this file in our S3 so it can be downloaded from there. That is consistent with what other tasks do.
```suggestion
    key: https://ansible-ci-files.s3.amazonaws.com/test/integration/targets/rpm_key/RPM-GPG-KEY.dag
```